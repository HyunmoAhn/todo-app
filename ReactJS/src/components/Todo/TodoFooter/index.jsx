import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const propTypes = {
  filter: PropTypes.string,
};
const defaultProps = {
  filter: 'all',
};

class TodoFooter extends React.Component {
  render() {
    const { filter } = this.props;
    const allClassName = cx({ selected: filter === 'all' });
    const activeClassName = cx({ selected: filter === 'active' });
    const completedClassName = cx({ selected: filter === 'completed' });

    return (
      <footer className="footer" style={{ display: 'block' }}>
        <span className="todo-count"><strong>1</strong> item left</span>
        <ul className="filters">
          <li><a href="#/" className={allClassName}>All</a></li>
          <li><a href="#/active" className={activeClassName}>Active</a></li>
          <li><a href="#/completed" className={completedClassName}>Completed</a></li>
        </ul>
        <button className="clear-completed" style={{ display: 'block' }}>
          Clear completed
        </button>
      </footer>
    );
  }
}

TodoFooter.propTypes = propTypes;
TodoFooter.defaultProps = defaultProps;

export default TodoFooter;
