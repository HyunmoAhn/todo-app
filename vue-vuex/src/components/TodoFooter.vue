<template>
  <footer class="footer">
    <span class="todo-count"><strong>{{ length }}</strong> item left</span>
    <ul class="filters">
      <li><a href="#/" :class="{ selected: filter === 'all' }">All</a></li>
      <li><a href="#/active" :class="{ selected: filter === 'active' }" >Active</a></li>
      <li><a href="#/completed" :class="{ selected: filter === 'completed' }">Completed</a></li>
    </ul>
    <button class="clear-completed" @click="deleteCompleted">
      Clear completed
    </button>
  </footer>
</template>
<script>
export default {
  name: 'todo-footer',
  props: {
    filter: String,
    length: Number,
    deleteCompleted: Function,
  },
};
</script>
