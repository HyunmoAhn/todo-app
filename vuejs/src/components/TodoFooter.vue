<template>
  <footer class="footer">
    <span class="todo-count"><strong>{{ length }}</strong> item left</span>
    <ul class="filters">
      <li><a href="#/" class="selected">All</a></li>
      <li><a href="#/active">Active</a></li>
      <li><a href="#/completed">Completed</a></li>
    </ul>
    <button class="clear-completed" @click="deleteCompleted">
      Clear completed
    </button>
  </footer>
</template>
<script>
export default {
  name: 'todo-footer',
  props: {
    length: Number,
    deleteCompleted: Function,
  },
};
</script>
