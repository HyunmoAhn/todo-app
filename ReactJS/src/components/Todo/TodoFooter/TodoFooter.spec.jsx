import React from 'react';
import { shallow } from 'enzyme';

import TodoFooter from './';

describe('<TodoFooter />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<TodoFooter />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClearCompletedItems when call handleClearCOmpletedItems', () => {
    const completedIds = ['mock-completed-id'];
    const onClearCompletedItems = jest.fn();
    const wrapper = shallow(
      <TodoFooter
        completedIds={completedIds}
        onClearCompletedItems={onClearCompletedItems}
      />,
    );

    wrapper.instance().handleClearCompletedItems();

    expect(onClearCompletedItems).toHaveBeenCalledTimes(1);
    expect(onClearCompletedItems).toHaveBeenCalledWith(completedIds);
  });
});
