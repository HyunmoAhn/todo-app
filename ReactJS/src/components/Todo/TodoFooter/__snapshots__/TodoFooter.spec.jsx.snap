// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoFooter /> should match to snapshot when render default 1`] = `
<footer
  className="footer"
  style={
    Object {
      "display": "block",
    }
  }
>
  <span
    className="todo-count"
  >
    <strong>
      0
    </strong>
     item left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className="selected"
        href="#/"
      >
        All
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/active"
      >
        Active
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/completed"
      >
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
  >
    Clear completed
  </button>
</footer>
`;
